types/react/index.d.ts

```diff
@@ -281,13 +282,18 @@ declare namespace React {
     // tslint:disable-next-line:no-empty-interface
     interface Component<P = {}, S = {}, SS = any> extends ComponentLifecycle<P, S, SS> { }
     class Component<P, S> {
+        constructor(props: Readonly<P>);
```

Comment:
Will remove the `Readonly` wrapper in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69140 since it doesn't seem to do anything.